--- usr/share/ldetect-lst/Cards+.legacy	2006-03-07 12:42:06.000000000 -0500
+++ usr/share/ldetect-lst/Cards+	2006-03-07 12:42:40.000000000 -0500
@@ -1931,7 +1931,7 @@
 SEE NVIDIA Legacy
 
 NAME NVIDIA GeForce3 (generic)
-SEE NVIDIA GeForce
+SEE NVIDIA Legacy
 
 NAME NVIDIA GeForce3 (xbox)
 DRIVER nvxbox
@@ -1939,7 +1939,7 @@
 LINE Option "HWCursor" "0"
 
 NAME NVIDIA GeForce4 (generic)
-SEE NVIDIA GeForce
+SEE NVIDIA Legacy
 
 NAME NVIDIA GeForce FX (generic)
 SEE NVIDIA GeForce
